dsp/transform/hilbert.go

```diff
@@ -0,0 +1,89 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"math/cmplx"
+
+	"gonum.org/v1/gonum/dsp/fourier"
+)
+
+// Implements an approximate Hilbert transform that allows the user to determine an
+// approximate analytical signal of a real signal, and determine the real 'envelope'
+// of a signal.
+//
+// The underlying implementation uses a DFT and IDFT transform, so usual methods for
+// speeding up DFT/IDFT are likely to apply here as well.
+type Hilbert struct {
+	fft  *fourier.CmplxFFT
+	work []complex128
+}
+
+// Return a new Hilbert transformer for signals of size N.
+func NewHilbert(n int) *Hilbert {
+	return &Hilbert{
+		fft:  fourier.NewCmplxFFT(n),
+		work: make([]complex128, n),
+	}
+}
+
+// Retrieve the length N of signals that are valid input for this Hilbert transform.
+func (h *Hilbert) Len() int {
+	return len(h.work)
+}
+
+// Compute the analytical signal of a real signal, and stores the result in the dst slice,
+// returning it.
+// If the dst slice is nil, a new slice will be created and returned. The dst slice
+// must be the same length as the input signal.
+func (h *Hilbert) AnalyticSignal(signal []float64, dst []complex128) []complex128 {
```

Comment:
```suggestion
func (h *Hilbert) AnalyticSignal(dst []complex128, signal []float64) []complex128 {
```
`dst` is conventionally the first parameter.